File: Triplens/src/main/java/com/org/Triplens/Services/UserService.java
package com.org.Triplens.Services;



import com.org.Triplens.entity.Users;
import com.org.Triplens.exception.NoUserFoundException;
import com.org.Triplens.exception.PasswordIncorrectException;

public interface UserService {
	boolean addUsers(String name,String password,String email);
	Users findUsers(String email) throws NoUserFoundException;
	boolean authenticate(String email,String password) throws NoUserFoundException, PasswordIncorrectException;
}

File: triplens-frontend/src/services/userService.js
import axios from "axios";

const API_BASE_URL = "http://localhost:8080/users";

export const authenticate = (email, password) => {
  return axios.post(
    `${API_BASE_URL}/authenticate`,
    null,
    {
      params: {
        email,
        password
      }
    }
  );
};

export const addUser = (user) => {
  return axios.post(
    `${API_BASE_URL}/addUser`,
    null,
    {
      params: {
        name: user.name,
        email: user.email,
        password: user.password
      }
    }
  );
};
